src/load_order.ts

```diff
@@ -211,35 +349,60 @@ export const internalDeserializeLoadOrder: VortexWrappedDeserializeFunc = async
   return Promise.resolve(newLO);
 };
 
-// Internal Deserializer so that we can have access to more good good information
-export const internalSerializeLoadOrder: VortexWrappedSerializeFunc = (
+
+const serializeLoadOrder: VortexWrappedSerializeFunc = (
   vortexApi: VortexApi,
-  loadOrder: VortexLoadOrder,
-) => {
-  vortexApi.log(`info`, `meow`);
+  vortexLoadOrder: VortexLoadOrder,
+): Promise<void> => {
+  vortexApi.log(`debug`, `${me}: Serialize`);
+
   const discoveryPath = getDiscoveryPath(vortexApi);
+
   if (discoveryPath === undefined) {
+    vortexApi.log(`error`, `${me}: Serialize: Game not found! (discoveryPath is undefined)`);
     return Promise.reject(new vortexUtil.NotFound(`Game Not Found.`));
   }
 
-  const modListPath = path.join(discoveryPath, 'r6', 'cache', 'modded', 'mods.json');
-  vortexApi.log(`debug`, `LO to write: `, loadOrder);
-  const mods: REDmodEntry[] = loadOrder.map((mod) => {
-    vortexApi.log(`debug`, `adding mod to write: `, mod);
-    const theRedEntry = mod.data.entry;
-    theRedEntry.enabled = mod.enabled;
-    return theRedEntry;
-  });
-  vortexApi.log(`debug`, `JSONifiable mods: `, mods);
-  const meowjson = JSON.stringify({ mods }, undefined, ' ');
-  vortexApi.log(`debug`, `stringy JSON being written: `, meowjson);
-  return fs.writeFileAsync(modListPath, meowjson, { encoding: 'utf8' });
+  vortexApi.log(`info`, `${me}: Serializing new load order from Vortex load order`, vortexLoadOrder);
+
+  const loadOrderEntries = pipe(
+    vortexLoadOrder,
+    map((entry: VortexLoadOrderEntry): LoadOrderEntry => ({
+      id: entry.id,
+      modId: entry.modId,
+      displayName: entry.name,
+      enabledInVortex: entry.enabled,
+      dirPath: entry.data?.dirPath || `unknown path`,
+      version: entry.data?.version || `unknown version`,
+    })),
+    toMutableArray,
+  );
+
+  const loadOrder: LoadOrder = {
```

Comment:
we don't need to keep any special meta infomation in the file. it should _just_ be a list of mods in order from 0-n where mods[0] is loaded first and mods[n] is loaded last.

Reply:
You always need metadata :) If not now, then when we realize the format needs to change.